### PR TITLE
record assessment activities for multiple choice assessment levels

### PR DESCRIPTION
**Pull request by davidsbailey**
### Background

Finishes https://codedotorg.atlassian.net/browse/LP-60

### Description

Store a log entry on every milestone post for multiple choice assessment levels. 

This meets the product goal because all levels in progressions named "Check Your Understanding" are assessments. 

### Alternative

One obvious alternative would be to use the existing activities table to store this data. I didn't want to do this for a few reasons:
* I want the `script_id` column, and adding it to the `activities` table could be difficult
* we may want to add more columns to this table in the future, which would again be difficult
* if we decide to **retire** the activities table, I don't want this new feature depending on all the complex async logic and preventing us from removing that logic
* if we decide to **re-enable** the activities table, it will be relatively cheap to keep this table operating separately, since this table will only record data for a small fraction of levels

### Deployment

The migration will be deployed ahead of this in a separate PR.

### Future work

In the future, it is possible we will want to enable this for other assessment level types besides multi. From a scale perspective, it should be ok if this is someday enabled for all assessment levels regardless of type, because assessments make up a relatively small fraction of our traffic.

### CODE REVIEW

**Review by maddiedierker — APPROVED** [review]
🥇 great tests! this looks good to me as long as the migration commit is pulled into its own PR to be deployed first

**Comment by davidsbailey** [comment]
Thank you, Maddie! Unfortunately, the "separate table" approach got voted down (see #29294). A very similar PR will be coming your way soon, writing to `activities` instead of `assessment_activities`.

**Comment by davidsbailey** [comment]
@madelynkasula , heads up that this approach is getting restored because #29356 was going to take too long.

@islemaster I've added a bit of code to purge data from this table, and verified locally that `level_source_id` get's wiped from this table for `user` when I run `DeleteAccountsHelper.new(bypass_safety_constraints: true).purge_user(user)`. Please let me know if there is anything else I need to do here regarding user deletion.

File: lib/cdo/delete_accounts_helper.rb [hunk 1]
```diff
@@ -82,6 +82,10 @@ def clean_level_source_backed_progress(user_id)
     updated_rows = GalleryActivity.where(user_id: user_id).update_all(level_source_id: nil)
     @log.puts "Cleaned #{updated_rows} GalleryActivity" if updated_rows > 0
 
+    @log.puts "Cleaning AssessmentActivity"
+    updated_rows = AssessmentActivity.where(user_id: user_id).update_all(level_source_id: nil)
+    @log.puts "Cleaned #{updated_rows} AssessmentActivity" if updated_rows > 0
```
Comment (islemaster) [inline lib/cdo/delete_accounts_helper.rb:87]: Sorry I'm late on the review for this.  I'd appreciate a test case for this in [DeleteAccountsHelperTest](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/helpers/delete_accounts_helper_test.rb#L446-L465).

**Pull request merged**